Pluggable tokenizer

iankronquist opened the pull request:
Loads and uses a pluggable tokenizer. Comes with a sane default tokenizer.

@willkg r?

iankronquist committed:
Load and use the tokenizer

iankronquist committed:
Load and use the tokenizer

iankronquist committed:
Import the split_tokenizer

iankronquist committed:
Import the split_tokenizer

iankronquist committed:
Remove tokenization from report generation

iankronquist committed:
Remove tokenization from report generation

iankronquist committed:
Add a default tokenizer

Splits on punctuation and whitespace, lower cases everything, and filter ''

iankronquist committed:
Add a default tokenizer

Splits on punctuation and whitespace, lower cases everything, and filter ''

iankronquist committed:
Add a BaseTokenizer

iankronquist committed:
Add a BaseTokenizer

iankronquist committed:
Add tests for SplitTokenizer

iankronquist committed:
Add tests for SplitTokenizer

iankronquist committed:
Clean up a bunch of old or unused files

Various scripts for sorting and reformatting data,
Move report.py into a scripts directory.
Remove tox.ini

willkg commented on spicedham/__init__.py:
```diff
@@ -59,6 +76,7 @@ def train(self, training_data, match):
         For instance, if you're filtering spam `match` will be True for spam,
         and False for a normal message.
         """
+        training_data = self.tokenizer.tokenize(training_data)
```
I think this makes the docstring incorrect.

willkg commented on spicedham/__init__.py:
```diff
@@ -67,6 +85,7 @@ def classify(self, classification_data):
         Calls each plugin's classify function and averages the results.
         `classification_data` is an iterable of strings.
         """
+        classification_data = self.tokenizer.tokenize(classification_data)
```
I think this makes the docstring incorrect.

willkg commented on spicedham/split_tokenizer.py:
```diff
@@ -0,0 +1,17 @@
+        is_not_blank = lambda x: x != ''
+        text = filter(is_not_blank, text)
+        lower_case = lambda x: x.lower()
+        text = map(lower_case, text)
```
I think I would have done the lower/blank pass in one list comprehension:

```
text = [token.lower() for token in text if token]
```

willkg commented on spicedham/split_tokenizer.py:
```diff
@@ -0,0 +1,17 @@
+from spicedham.tokenizer import BaseTokenizer
+
+class SplitTokenizer(BaseTokenizer):
+    
```
This could use a docstring about what it does. I think I'd take the docstring for the tokenize method and move it up to the SplitTokenizer class.

willkg commented on spicedham/gottaimportthemall.py:
```diff
@@ -4,5 +4,7 @@
 try:
     from spicedham.sqlalchemywrapper import SqlAlchemyWrapper
+    # As part of the default configuration
+    import spicedham.split_tokenizer
```
If importing SqlAlchemyWrapper kicks up an import error, it'll never reach this line. Further, I don't see why the SplitTokenizer would kick up an import error, so I'd move it outside the try/except block.

willkg commented:
I had a handful of issues that should get looked at and then I can do another review pass.

iankronquist committed:
Import error should not block tockenizer

iankronquist committed:
Import error should not block tockenizer

iankronquist committed:
Hoist the docstring from the method into the class

iankronquist committed:
Hoist the docstring from the method into the class

iankronquist committed:
Make mapping & filtering into a list comprehension

iankronquist committed:
Make mapping & filtering into a list comprehension

iankronquist committed:
Update docstrings to mention the tokenizer

iankronquist committed:
Update docstrings to mention the tokenizer

iankronquist commented:
@willkg Changed as you suggested.

iankronquist commented:
@willkg r?

iankronquist committed:
Add tox.ini back in

iankronquist committed:
Fix #21

When the classifier isn't trained you shouldn't be able to use it.
Even in tests.

iankronquist committed:
Fix docstring

I changed my mind in the middle and forgot to change the docstring

iankronquist committed:
Use a custom exception, not built-in value error

iankronquist committed:
Merge pull request #30 from iankronquist/tests-2.6-fixes

Fix this python 2.6 assertion nonsense

willkg commented on spicedham/split_tokenizer.py:
```diff
@@ -0,0 +1,14 @@
+from re import split
```
I wouldn't do this. "split" is pretty common in other contexts, so it's better to do:

```
import re
```

and then call it with `re.split`.

willkg commented:
Other than that nit about `re.split`, this looks good to me. r+

iankronquist committed:
Merge pull request #29 from iankronquist/untrained-exception

Untrained exception

iankronquist committed:
Merge pull request #24 from iankronquist/cleanup-10

Clean up a bunch of old or unused files

iankronquist committed:
Be clear that we're using re.split

iankronquist committed:
Be clear that we're using re.split

iankronquist commented:
Rebasing...

willkg commented:
This looks good to me! r+

iankronquist committed:
Rewrite tests to rely on mock less

Also test _load_tokenizer

iankronquist committed:
Merge branch 'parser' of github.com:iankronquist/spicedham into parser
